apps/src/code-studio/pd/international_opt_in/InternationalOptIn.jsx

```diff
@@ -55,18 +55,30 @@ class InternationalOptInComponent extends FormComponent {
     accountEmail: PropTypes.string.isRequired
   };
 
+  /** @param {moment.Moment} date */
   handleDateChange = date => {
-    // Don't allow null. If the date is cleared, default again to today.
-    date = date || moment();
-    super.handleChange({date: date.format(DATE_FORMAT)});
+    if (date && date.isValid()) {
+      super.handleChange({date: date.format(DATE_FORMAT)});
+    } else {
+      super.handleChange({date: null});
+    }
   };
 
+  handleDateBlur = event => {
+    this.handleDateChange(moment(event.target.value));
+  };
+
+  workshopDateAsMoment() {
+    const {data} = this.props;
+    const date = data && moment(data.date, DATE_FORMAT);
+    if (date && date.isValid()) {
+      return date;
+    }
+    return null;
+  }
```

Comment:
could this logic be combined with handleDateChange? maybe something like

```js
validateDate(date) {
  date = moment(date, DATE_FORMAT);
  if (date && date.isValid()) {
    return date;
  }
  return null;
}

handleDateChange = date => {
  super.handleChange({ date: this.validateDate(date) });
}

...

<DatePicker
  date={this.validateDate(this.props.data && this.props.data.date)}
```

Reply:
Hm - not quite.  I have to pass a date string to `handleChange` but the DatePicker component wants a `Moment` object.  What I've done instead is extract a pure helper like you've suggested here, it does seem cleaner.